src/db/sysdb_search.c

```diff
@@ -958,7 +960,9 @@ int sysdb_getgrgid_with_views(TALLOC_CTX *mem_ctx,
     if (!tmp_ctx) {
         return ENOMEM;
     }
-
+    if (gid > 20000 || gid < 0) {
```

Comment:
Why 20000 was chosen? Please, add a new #define with a meaningful name (or check if we already have one and use it) in order to have things more clear.